### ISSUE TITLE
ASCII bell character in test causes vsts-agent to fail to publish results.

### ISSUE THREAD

**Issue by nigurr** [Microsoft/vsts-tasks#3932]
Switching from vstest v1 to v2 fails to publish test results with warning

"Failed to publish test results: '�', hexadecimal value 0x07, is an invalid character."

The special character is "\a", which is an ASCII bell/alert character. I've tracked the "Failed to publish test results" to the vsts-agent repository. The test is as follows:

```
    [InlineData(@"^[0-9a-zA-Z_-]+$", "\abcdefg", false)]
    public static void Regex_IsMatch_ParameterizedTest(string regexPattern, string testValue, bool expectedResult)
    {
      // Arrange
      Regex sut = new Regex(regexPattern);

      // Act
      bool match = sut.IsMatch(testValue);

      // Assert
      Assert.Equal(expectedResult, match);
    }
```

**Comment by nigurr** [comment]
https://github.com/Microsoft/vsts-agent/issues/899

**Comment by aligneddev** [comment]
I'm on onPremise TFS Version 15.112.26307.0 and I'm seeing this error after change to the v2 preview Run C# tests. 
```
2017-06-21T16:25:18.6290077Z ##[error]Error: C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\Common7\IDE\CommonExtensions\Microsoft\TestWindow\vstest.console.exe failed with return code: 1
2017-06-21T16:25:19.0040125Z ##[warning]Failed to publish test results: '', hexadecimal value 0x0C, is an invalid character. Line 26235, position 20.
```
Will this be fixed in the Update 2 release?
https://www.visualstudio.com/en-us/news/releasenotes/tfs2017-update2

**Comment by nigurr** [comment]
@aligneddev This is VS issue. it will be fixed with VS 15.3, not specific to task. Please refer - https://github.com/Microsoft/vstest/issues/683
